Fix PluginWrapper bug and cleanup Javadoc placeholders

shbhmexe opened the pull request:
This PR fixes a logic bug in an unused, deprecated method in `PluginWrapper` and updates placeholder version numbers in Javadoc deprecation annotations.

### Testing done

- Verified compilation with `mvn compile -pl core -q` - PASSED
- Manual verification of Javadoc changes

### Proposed changelog entries

- Fix incorrect variable passed in deprecated `PluginWrapper.setOptionalDependants()` method.

### Proposed changelog category

/label skip-changelog

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.

### Desired reviewers

@jenkinsci/core-pr-reviewers

shbhmexe committed:
Fix synchronization, PluginWrapper bug, and cleanup Javadoc

- Fix inconsistent synchronization in Fingerprint.java on 'usages' field by synchronizing on 'this'.
- Fix logic bug in PluginWrapper.java where setOptionalDependants passed incorrect field instead of parameter.
- Replace Boolean.getBoolean with SystemProperties.getBoolean in Jenkins.java and JnlpSlaveRestarterInstaller.java for best practices.
- Update placeholder version numbers (@deprecated and @since) in StreamTaskListener, StreamBuildListener, and JNLPLauncher.

Signed-off-by: shbhmexe <shubhushukla586@gmail.com>

shbhmexe committed:
Merge branch 'master' into fix/jenkins-core-cleanup-and-sync

MarkEWaite commented on core/src/main/java/hudson/model/StreamBuildListener.java:
```diff
@@ -48,7 +48,7 @@ public StreamBuildListener(File out, Charset charset) throws IOException {
 
     /**
-     * @deprecated as of TODO
+     * @deprecated as of 2.328
```
The deprecation was added in 2.329 as far as I can tell, commit 3b29e9d460381f83eba9e939f6053b8336d946f4
```suggestion
     * @deprecated as of 2.329
```

shbhmexe replied:
Thank you for the correction! I will commit this suggestion. The version should indeed be 2.329.

MarkEWaite commented on core/src/main/java/hudson/PluginWrapper.java:
```diff
@@ -253,7 +253,7 @@ public void setOptionalDependents(@NonNull Set<String> optionalDependents) {
     @Deprecated
     public void setOptionalDependants(@NonNull Set<String> optionalDependents) {
-        setOptionalDependents(dependents);
+        setOptionalDependents(optionalDependents);
```
Very nice catch!  As far as I can tell from a search of the jenkinsci GitHub organization, there are no consumers of this deprecated method.  The code change is correct.

MarkEWaite reviewed (changes requested):
This seems like a combination of multiple changes into a single pull request.  Please limit the scope of changes in pull requests so that reviewers are more likely to review the change and so that they can be clearly described.

The change of argument for the `setOptionalDependants` method is correct.  Since the method is unused, the change will likely have no effect, but the change is correct.

The updates of the Javadoc comments generally mention a version prior to the deprecation.  We list the version that first deprecated the method.   Since there are very few of these cases, it is probably not worth the effort to update the `update-since-todo.py` script that automates the replacement of TODO with a release date.

The change of synchronization in the Fingerprint class does not have enough justification for the change and it lacks testing.  If that justification and testing are added, it should be a separate commit.

The change from Boolean.getBoolean() to SystemProperties.getBoolean does not have enough justification for the change.  Specifically, there is no mention of a reason why the ServletContext `init` context should be consulted for the property in addition to checking the system properties.  If that justification and sufficient testing are added, it should be a separate commit

shbhmexe commented on core/src/main/java/hudson/model/Fingerprint.java:
```diff
@@ -1036,7 +1036,7 @@ protected Object readResolve() {
 
     void addWithoutSaving(@NonNull String jobFullName, int n) {
-        synchronized (usages) { // TODO why not synchronized (this) like some, though not all, other accesses?
+        synchronized (this) {
```
You're absolutely right @MarkEWaite. I apologize for not providing sufficient justification.

shbhmexe committed:
Update core/src/main/java/hudson/model/StreamBuildListener.java

Co-authored-by: Mark Waite <mark.earl.waite@gmail.com>

shbhmexe commented on core/src/main/java/hudson/slaves/JNLPLauncher.java:
```diff
@@ -185,7 +185,7 @@ public void launch(SlaveComputer computer, TaskListener listener) {
 
     /**
-     * @deprecated as of 1.XXX
+     * @deprecated as of 2.0
```
Thank you for finding the correct version! I will commit this suggestion. The Javadoc was indeed added in version 2.2.

shbhmexe committed:
Update core/src/main/java/hudson/slaves/JNLPLauncher.java

Co-authored-by: Mark Waite <mark.earl.waite@gmail.com>

shbhmexe committed:
Update core/src/main/java/hudson/util/StreamTaskListener.java

Co-authored-by: Mark Waite <mark.earl.waite@gmail.com>

shbhmexe committed:
Update core/src/main/java/jenkins/model/Jenkins.java

Co-authored-by: Mark Waite <mark.earl.waite@gmail.com>

shbhmexe commented on core/src/main/java/jenkins/slaves/restarter/JnlpSlaveRestarterInstaller.java:
```diff
@@ -36,7 +37,7 @@ public class JnlpSlaveRestarterInstaller extends ComputerListener implements Ser
      * To force installer to run on all agents, set this system property to true.
      */
-    private static final boolean FORCE_INSTALL = Boolean.getBoolean(JnlpSlaveRestarterInstaller.class.getName() + ".forceInstall");
+    private static final boolean FORCE_INSTALL = SystemProperties.getBoolean(JnlpSlaveRestarterInstaller.class.getName() + ".forceInstall");
```
You're absolutely right @MarkEWaite. Similar to the SHOW_STACK_TRACE case, I don't have evidence that FORCE_INSTALL property is configured via ServletContext.

I will accept this suggestion and revert to using Boolean.getBoolean().

Thank you for catching this - these changes were made without proper justification. I'll be more careful about proposing "best practice" changes without concrete use cases.

shbhmexe committed:
Update core/src/main/java/jenkins/slaves/restarter/JnlpSlaveRestarterInstaller.java

Co-authored-by: Mark Waite <mark.earl.waite@gmail.com>

shbhmexe committed:
Update Fingerprint.java

shbhmexe committed:
Revert JnlpSlaveRestarterInstaller SystemProperties change per reviewer feedback

shbhmexe commented:
Thanks @MarkEWaite! I've updated the PR description to reflect only the current changes (PluginWrapper fix + Javadoc updates). Removed mention of Fingerprint and SystemProperties changes.

shbhmexe committed:
Merge branch 'master' into fix/jenkins-core-cleanup-and-sync

MarkEWaite reviewed (approved):
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback.

/label ready-for-merge

shbhmexe committed:
Merge branch 'master' into fix/jenkins-core-cleanup-and-sync
